### PR TITLE
Allow multiple test execution runs

### PR DESCRIPTION
**Pull request by omar-selo**
## Description

Includes the following changes:
- Drop unneeded review columns from Test Execution (now that the previous migration seems to have been safe)
- Remove uniqueness constraint on Test Executions to allow multiple to exist in case of reruns
- Remove automatic approvals (needs a separate revisit)
- Show multiple test execution runs on the frontend
- Replace test execution status filter with last test execution status filter (required a small refactor)
- Group previous test results by artefact version
- Reverse order of previous results and add the current result as large icon https://warthogs.atlassian.net/browse/RTW-398
- Fix issues and improve performance of previous results query https://warthogs.atlassian.net/browse/RTW-408 

Notes:
- Test result history now shows history including previous runs
- Environment expandable has a test execution status icon next to it for the last run
- Among the changes to the query that fetches previous results, it now doesn't only consider latest builds. This is because we've learned during the sprint that newer builds should be considered as reruns. There are more changes that need to be done to get there, but this is a first step

## Resolved issues

Resolves https://warthogs.atlassian.net/browse/RTW-387
Resolves https://warthogs.atlassian.net/browse/RTW-398
Resolves https://warthogs.atlassian.net/browse/RTW-408

## Web service API changes

Nothing.

## Tests

Added some automated tests. And manually tested the frontend.

[Screencast from 2024-11-22 17-31-50.webm](https://github.com/user-attachments/assets/7c9dcda0-a9a7-41ff-9357-315cd8c3c65c)

### CODE REVIEW

**Comment by omar-selo** [comment]
Converted PR to draft as I still need to group historic test results by artefact version.

File: frontend/lib/ui/artefact_page/test_result_expandable.dart [hunk 1]
```diff
@@ -73,24 +78,81 @@ class _TestResultOutputExpandable extends StatelessWidget {
...
+    }
+
     return Row(
-      children: previousResults
-          .map(
-            (e) => InkWell(
-              onTap: () => navigateToArtefactPage(context, e.artefactId),
+      children: statusGroups.entries
+          .mapIndexed<Widget>(
+            (groupIndex, entry) => InkWell(
```
Comment (andrejvelichkovski) [inline frontend/lib/ui/artefact_page/test_result_expandable.dart:123]: We seem to have quite a few elements inside this `InkWell` widget, perhaps we can move it to a separate private method that returns a `PreviousResultStatusGroup` widget

**Review by andrejvelichkovski — APPROVED** [review]
I went through all the changes, starting from the database, then the backend to support it, and finally on the frontend, and everything seems reasonable to me. I think you covered everything that I managed to think of.

I know we are introducing some limitations with this change, such as the the summary of test executions (artefact environments) that we used to have will now be gone, but I think this is all known in the implementation plan and we will provide the equivalent after we complete all phases.

I left a few mainly style comments. Thanks for all the work on this!

File: backend/migrations/versions/2024_10_21_1035-063e32aac8db_allow_multiple_testexecution_runs.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    remove_test_execution_runs_keeping_latest()
+
+    op.create_unique_constraint(
+        "test_execution_artefact_build_id_environment_id_key",
+        "test_execution",
+        ["artefact_build_id", "environment_id"],
+    )
+
+
+def remove_test_execution_runs_keeping_latest():
```
[the remark this answers is not in the document]
Comment (omar-selo) [reply backend/migrations/versions/2024_10_21_1035-063e32aac8db_allow_multiple_testexecution_runs.py:38]: I sure hope we don't have to do that. I've tried it locally and it worked fine at least for what I've tried.

**Comment by omar-selo** [comment]
> I know we are introducing some limitations with this change, such as the the summary of test executions (artefact environments) that we used to have will now be gone, but I think this is all known in the implementation plan and we will provide the equivalent after we complete all phases.

Good catch. I actually missed this. I've added it back using the latest test execution result for each environment.

**Review by andrejvelichkovski — APPROVED** [review]
Thanks for addressing the comments! I think we are ready to merge the changes now.

**Pull request merged**